rust/worker/src/blockstore/arrow_blockfile/blockfile.rs

```diff
@@ -1 +1,557 @@
+use super::super::types::{Blockfile, BlockfileKey, Key, KeyType, Value, ValueType};
+use super::block::{BlockError, BlockState};
+use super::provider::ArrowBlockProvider;
+use super::sparse_index::SparseIndex;
+use crate::blockstore::arrow_blockfile::block::delta::BlockDelta;
+use crate::blockstore::BlockfileError;
+use crate::errors::ChromaError;
+use parking_lot::Mutex;
+use std::sync::Arc;
+use thiserror::Error;
+use uuid::Uuid;
+
 pub(super) const MAX_BLOCK_SIZE: usize = 16384;
+
+#[derive(Clone)]
+pub(crate) struct ArrowBlockfile {
+    key_type: KeyType,
+    value_type: ValueType,
+    block_provider: ArrowBlockProvider,
+    sparse_index: Arc<Mutex<SparseIndex>>,
+    transaction_state: Option<Arc<TransactionState>>,
+}
+
+struct TransactionState {
+    block_delta: Mutex<Vec<BlockDelta>>,
+    new_sparse_index: Mutex<Option<Arc<Mutex<SparseIndex>>>>,
```

Comment:
Nit: can we rename this to just `sparse_index`? Every time I read it I think we're creating a new sparse index

Reply:
Yeah sure, the reason i did call it this is its the "new" index pending the transaction commit